src/plugins/intel_cpu/src/cpu_memory.cpp

```diff
@@ -115,6 +116,13 @@ void Memory::redefineDesc(MemoryDescPtr desc) {
         IE_THROW() << "Can not reset descriptor, memory upper bound is unknown.";
     }
 
+    // TODO: how elegantly
+    const auto memMngr = getMemoryMngr();
+    auto outMemMngr = std::dynamic_pointer_cast<OutputMemoryMngr>(memMngr);
+    if (outMemMngr != nullptr) {
+        outMemMngr->setMemDesc(desc);
+    }
```

Comment:
As we discussed, this should not work that way. The memory descriptor may not reflect the memory size to be allocated. Please consider inPlace concat chains. As it was agreed, the reallocation should happen using the size requested from the memory manager, via `OutputMemoryMngr::resize` wrapped call.